playwright-gh-pages/upload-report-artifacts/action.yml

```diff
@@ -0,0 +1,94 @@
+name: 'Upload report and test summary to GitHub artifact'
+description: 'Uploads report and test summary as a GitHub artifact. Can be used together with the publish-report-pages action to publish the reports to GitHub Pages.'
+author: 'frontend@grafana'
+
+outputs:
+  artifact:
+    description: "The path to the uploaded artifact."    
+    value: ${{ steps.set-artifact-name.outputs.artifact }}
+
+
+inputs:
+  github-token:
+    description: 'Token for the repository. Can be passed in using `{{ secrets.GITHUB_TOKEN }}`.'
+    required: true
+    default: ${{ github.token }}
+  test-outcome:
+    description: 'Outcome of the test step. For example `{{ steps.run-tests.outcome }}`.'
+    required: true
+  grafana-image:
+    description: 'Grafana image used in the test. Default is `{{ matrix.GRAFANA_IMAGE.NAME }}`.'
+    default: ${{ matrix.GRAFANA_IMAGE.NAME }}
+    required: true
+  grafana-version:
+    description: 'Grafana version used in the test. Default is `{{ matrix.GRAFANA_IMAGE.VERSION }}`.'
+    default: ${{ matrix.GRAFANA_IMAGE.VERSION }}
+    required: true
+  artifact-prefix:
+    description: 'Pattern to prefix the artifact with. Default is "playwright-report-".'
+    required: false
+    default: 'playwright-report-'
+  upload-successful-reports:
+    description: 'Whether to upload the report if all tests were successful.'
+    required: false
+    default: false
+  report-dir:
+    description: 'Directory in which the report is stored. Default is "playwright-report".'
+    required: true
+    default: playwright-report
+  plugin-name:
+    description: 'Name of the plugin being tested. Useful in mono-repos when multiple plugins are tested generating multiple reports.'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Add parent directory with a name unique to the test setup
+      shell: bash
+      id: add-parent-dir
+      run: |
+        parent_dir=${{ inputs.grafana-image }}-${{ inputs.grafana-version }}
+        if [[ -n "${{ inputs.plugin-name }}" ]]; then
+          parent_dir="${{ inputs.plugin-name }}-${{ inputs.grafana-image }}-${{ inputs.grafana-version }}"
+        fi
+        echo $parent_dir
+        mv ${{ inputs.report-dir }} $parent_dir
+        mkdir -p playwright-report
+        mv $parent_dir playwright-report/
+        ls playwright-report
+        echo "::set-output name=parent-dir::$parent_dir"
```

Comment:
`set-output` is deprecated and `GITHUB_OUTPUT` shoule be used instead: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/